Analysis/Utkscan/core/include/SegmentDetector.hpp

```diff
@@ -0,0 +1,119 @@
+/** @file SegmentDetector.hpp
+ * @brief  Basic SegmentDetector Class (Similar to BarDetector, but without the HighRes Timing stuff). Originally written as MtasSegment but now generalized.
+ * @authors T.T. King, T. Ruland, B.C. Rasco
+ * @date 03/25/2022
+ */
+#ifndef PAASS_SegmentDetector_H
+#define PAASS_SegmentDetector_H
+
+#include "RawEvent.hpp"
+#include "Globals.hpp"
+
+class SegmentDetector {
+   public:
+    /** Constructor */
+    SegmentDetector() {
+        segFront_ = nullptr;
+        segBack_ = nullptr;
+    };
+    /** Destructor */
+    ~SegmentDetector() = default;
+
+    virtual ChanEvent* GetSegFront() {
+        return segFront_;
+    };
+    virtual ChanEvent* GetSegBack() {
+        return segBack_;
+    };
+
+    virtual void SetSegFront(ChanEvent* a) {
+        segFront_ = a;
+    };
+    virtual void SetSegBack(ChanEvent* a) {
+        segBack_ = a;
+    };
+
+    virtual void SetPixieRev(string &a) {
+        PixieRev = a;
+    };
+
+    virtual string GetPixieRev(){
```

Comment:
Seems like this should probably be const noexcept

Reply:
completed